src/main.rs

```diff
@@ -146,22 +162,30 @@ fn substitute_source(
 ) -> anyhow::Result<()> {
     static TEMPLATE_REGEX: OnceLock<regex::Regex> = OnceLock::new();
     let regex = TEMPLATE_REGEX.get_or_init(|| regex::Regex::new(r"%\{(.*?)\}").unwrap());
-    while let Some(captures) = regex.captures(manifest) {
-        let (Some(full), Some(capture)) = (captures.get(0), captures.get(1)) else {
-            continue;
-        };
-        let template = capture.as_str();
-        let (template_key, template_value) = template.split_once('=').with_context(|| {
-            format!("invalid template '{template}'(template should be in the form $KEY=$VALUE)")
-        })?;
-        let path = components
-            .get(template_value)
-            .with_context(|| format!("'{template_value}' is not a known component"))?;
-        let component_file = format!("{template_value}.wasm");
-        std::fs::copy(path, test_archive.join(&component_file))?;
-        if template_key.trim() == "source" {
-            manifest.replace_range(full.range(), &component_file);
+    'outer: loop {
+        for captures in regex.captures_iter(manifest) {
+            let (Some(full), Some(capture)) = (captures.get(0), captures.get(1)) else {
+                continue;
+            };
+            let template = capture.as_str();
+            let (template_key, template_value) = template.split_once('=').with_context(|| {
+                format!("invalid template '{template}'(template should be in the form $KEY=$VALUE)")
+            })?;
+            let (template_key, template_value) = (template_key.trim(), template_value.trim());
+            if "source" == template_key {
+                let path = components
+                    .get(template_value)
+                    .with_context(|| format!("'{template_value}' is not a known component"))?;
+                let component_file = "component.wasm";
+                std::fs::copy(path, test_archive.join(component_file))?;
```

Comment:
Is this only evaluating the source under `component.test`? I think we need to update the `request-shape` app's `spin.toml`. Right now, i believe that the source file is not being replaced and the shim tests are failing with:
```sh
thread 'main' panicked at conformance-tests/src/main.rs:38:10:
called `Result::unwrap()` on an `Err` value: failed to copy file '/tmp/conformance-tests/request-shape/component.wasm' to temporary directory
```

Reply:
The tests work for me in the shim. Are you updating the shim conformance test binary to use the last version of the conformance-tests library?

Reply:
I was, but I'll give it another go